### ISSUE TITLE
Incorrect "CS0252 Possible unintended reference comparison" for generated record == operator overload

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#49302]
**Version Used**: Both 16.8.0 and 16.9-p1

`a.B` and `b` are both expressions of type `RecordB`:

![image](https://user-images.githubusercontent.com/8040367/98880142-18185500-2455-11eb-9488-fcdecbf94fae.png)

If I keep leaving and hovering again, I'll intermittently see the correct operator overload defined on `RecordB`:

![image](https://user-images.githubusercontent.com/8040367/98880224-44cc6c80-2455-11eb-8fa7-a82c81ac7569.png)

### Repro

ℹ If the repro doesn't appear, erase a character from the `a.B == b` expression, wait a second for the IDE to reanalyze, then type the character back in and wait a second again.

Source file 1:
```cs
record RecordA(RecordB B);

record RecordB(int C);
```

Source file 2 (must be separate):
```cs
class C
{
    void M(RecordA a, RecordB b)
    {
        _ = a.B == b;
    }
}
```

Project:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net5.0</TargetFramework>
    <LangVersion>9.0</LangVersion>
  </PropertyGroup>

</Project>
```

**Comment by jaredpar** [comment]
Moving to IDE team. I can confirm the behavior that @jnm2 is seeing here but it does not happen when building from the command line. It's only happening in the IDE experience. That's a bit odd.

**Comment by CyrusNajmabadi** [comment]
we only get these diagnostics from the compiler though @jaredpar :)

Could it be an issue with incremental builds?  potentially an issue where the compiler is using the wrong equality construct internally (i.e. comparing symbols by reference, instead of by value)?

**Comment by jaredpar** [comment]
> we only get these diagnostics from the compiler though @jaredpar :)

I agree. That's why I noted "that's a bit odd". I can't really conceive how one gives the warning and the other does not. If there were a delay (shows up in compiler then later in the IDE) yeah that I get. But not showing up in compiler but showing up in the IDE is perplexing. 

The one rationalization I could come up with is if there was a warning which was disabled in the compiler but some how enabled when you host us in the IDE. Not sure that fits here.

> Could it be an issue with incremental builds? potentially an issue where the compiler is using the wrong equality construct internally (i.e. comparing symbols by reference, instead of by value)?

Hmm, that's an interesting theory. Possibly supported by the idea that you need separate files to observe this.

**Comment by sschultze** [comment]
This is a bug I stumbled upon too. The compiler doesn't emit the warning, but the IDE (Visual Studio Code) does, even after restarting.

**Comment by Tratcher** [comment]
@jaredpar @CyrusNajmabadi I'm also hitting this now. Let me know if you want to collect any diagnostics.

**Comment by jcouv** [comment]
Fix is in progress. Aiming for 16.9.

**Comment by jcouv** [comment]
Fix was merged for 16.9p4. Closing
